server/entity/animation/animation.go

```diff
@@ -0,0 +1,61 @@
+package animation
+
+// Animation represents an animation & controller that may be attached to an entity.
+// Animations and controllers must be defined in a resource pack
+type Animation struct {
+	name, state, controller string
+	stopCondition           string
+}
+
+// New returns a new animation that can be attached to an entity. By default no controller or state is sent to the viewer.
+// To add a state and controller use WithController and WithState respectively.
+func New(animationName string) Animation {
+	return Animation{
+		name:          animationName,
+		state:         "",
+		controller:    "",
+		stopCondition: "",
+	}
+}
+
+// Name returns the name of the animation to be played
+func (a Animation) Name() string {
+	return a.name
+}
+
+// WithController sets the controller with the specified state.
+// The controller must be added in a resource pack
+func (a Animation) WithController(controller string) Animation {
+	a.controller = controller
+	return a
+}
+
+// Controller returns the name of the controller being used. Controller returns an empty string if
+// no controller was previously set
```

Comment:
Don't think it's necessary to mention the empty string here, also purely just for OCD reasons could you move Controller(), State() and StopCondition() methods to be before their respective With...() methods?